src/main/java/com/github/invictum/reportportal/ReportIntegrationConfig.java

```diff
@@ -11,8 +11,10 @@
  */
 public class ReportIntegrationConfig {
 
-    static final String COMMUNICATION_DIR_KEY = "serenity.rp.communication.dir";
-    static final String MODULES_COUNT_KEY = "serenity.rp.modules.count";
+    public static final String COMMUNICATION_DIR_KEY = "serenity.rp.communication.dir";
+    public static final String MODULES_COUNT_KEY = "serenity.rp.modules.count";
+    public static final String FAILSAFE_RERUN_KEY = "failsafe.rerunFailingTestsCount";
+    public static final String SUREFIRE_RERUN_KEY = "surefire.rerunFailingTestsCount";
```

Comment:
The main purpose of `ReportIntegrationConfig` is to have a common place for setting configuration programmatically. This adds flexibility to define different approaches for configuration. E. g. config from environment properties or command line arguments.

I guess you are used `COMMUNICATION_DIR_KEY` and `MODULES_COUNT_KEY` properties as an example. Unfortunately those properties not really good candidates for it. They have special meaning and have to be defined before DI bootstrap phase, there is why they forced to be defined as system properties. Properties introduced by new functionality don't have such requirements, so it is better to implement they as usual variables isolated in configuration instance. Use `usePreset` or `useClassNarrativeFormatter` as an examples.

Reply:
I use system properties, because test retries mechanism implemented in maven plugins: [FailSafe](https://maven.apache.org/surefire/maven-failsafe-plugin/examples/rerun-failing-tests.html) and [Surefire](https://maven.apache.org/surefire/maven-surefire-plugin/examples/rerun-failing-tests.html). This plugins have configuration logic via system properties and i just use it.